### PR TITLE
Full Support BigPicture(BannerMode) and Fixed issue of largeIcon using url  [Android]

### PR DESCRIPTION
**Pull request by admin2684**
Now this library (react-native-push-notification) can support these features:
   - BigPicture with "imageUrl" key in Json notification data 
   - Finally you can use https:// , http://  in largeIcon but you must using  "largeIconUrl" key in Json notification data (Please remove largeIcon key )

IMAGE 2020-05-17 09:14:37

Note: These changes has not effect on other features (Without remove other features)

Thanks to @tidianeb5

@Dallas62 Please merge this pull request

### CODE REVIEW

**Comment by lukhol** [comment]
I'm really waiting for this one. Hope it will be merged.

File: android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java [hunk 1]
```diff
@@ -698,6 +837,9 @@ private void checkOrCreateChannel(NotificationManager manager, String channel_id
                         .build();
 
                 channel.setSound(soundUri, audioAttributes);
+                channel.setDescription(this.config.getChannelDescription());
+                channel.enableLights(true);
+                channel.enableVibration(true);
```
Comment (Dallas62) [inline android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java:842]: This code is duplicate with L829-831.

File: android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java [hunk 2]
```diff
@@ -410,6 +449,7 @@ public void sendToNotificationCentre(Bundle bundle) {
                 notification.setVibrate(vibratePattern);
             }
 
+            soundUri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
```
Comment (Dallas62) [inline android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java:452]: This line introduce a regression by forcing sounds.

File: android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java [hunk 3]
```diff
@@ -145,12 +160,15 @@ public void sendNotificationScheduledCore(Bundle bundle) {
...
+            getAlarmManager().setExact(AlarmManager.RTC_WAKEUP, fireDate, pendingIntent);
         } else {
             getAlarmManager().set(AlarmManager.RTC_WAKEUP, fireDate, pendingIntent);
         }
     }
 
-    public void sendToNotificationCentre(Bundle bundle) {
+    public void sendNotificationWithImage(Bundle bundle, Bitmap image, Bitmap largeIconImage) {
+        Log.d(TAG, "sendNotificationWithImage: bitmao log: " + image);
+        Log.d(TAG, "sendNotificationWithImage: bitmao log: " + largeIconImage);
```
Comment (Dallas62) [inline android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java:171]: A `LOG_TAG` constant already exist.

File: android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java [hunk 4]
```diff
@@ -57,6 +70,8 @@
     private static final long ONE_HOUR = 60 * ONE_MINUTE;
     private static final long ONE_DAY = 24 * ONE_HOUR;
 
+    private String TAG = "RNPushNotificationHelper";
```
Comment (Dallas62) [inline android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java:73]: A `LOG_TAG` constant already exist.

File: android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java [hunk 5]
```diff
@@ -145,12 +160,15 @@ public void sendNotificationScheduledCore(Bundle bundle) {
             } else {
                 getAlarmManager().setExact(AlarmManager.RTC_WAKEUP, fireDate, pendingIntent);
             }
+            getAlarmManager().setExact(AlarmManager.RTC_WAKEUP, fireDate, pendingIntent);
```
Comment (Dallas62) [inline android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java:163]: This code is duplicate, result is:
```java
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
            if(allowWhileIdle && Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
                getAlarmManager().setExactAndAllowWhileIdle(AlarmManager.RTC_WAKEUP, fireDate, pendingIntent);
            } else {
                getAlarmManager().setExact(AlarmManager.RTC_WAKEUP, fireDate, pendingIntent);
            }
            getAlarmManager().setExact(AlarmManager.RTC_WAKEUP, fireDate, pendingIntent);
        } else {
            getAlarmManager().set(AlarmManager.RTC_WAKEUP, fireDate, pendingIntent);
        }
```

File: android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java [hunk 6]
```diff
@@ -330,6 +349,26 @@ public void sendToNotificationCentre(Bundle bundle) {
             }
 
             notification.setStyle(new NotificationCompat.BigTextStyle().bigText(bigText));
+            if (largeIconImage != null) {
+                notification.setLargeIcon(largeIconImage);
+            } else {
+                if (largeIconResId != 0 && (largeIcon != null || Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP)) {
+                    notification.setLargeIcon(largeIconBitmap);
+                }
+            }
```
Comment (Dallas62) [inline android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java:358]: This code is also duplicate with L340-342

**Comment by Dallas62** [comment]
Thanks for your contribution, this helped to implement a solution to this. You can test the version on `dev` for this feature.
https://github.com/zo0r/react-native-push-notification/tree/dev

**Pull request closed without merging**